prow/hook/events.go

```diff
@@ -51,6 +51,7 @@ var (
 )
 
 func (s *Server) handleReviewEvent(l *logrus.Entry, re github.ReviewEvent) {
+	defer s.wg.Done()
```

Comment:
This doesn't actually wait for the event to be fully handled. Below we create a new goroutine for each handler, but the handler goroutines aren't being waited for. Those are the threads that need to be counted with the wait group.

Reply:
each `wg.Done()` also needs to be correctly mapped to `wg.Add()`, so when fixing this each of the `s.wg.Add()`s should probably be moved out of the switch statement and into the handlers / at the site of each async handler spawned...

side nit: goroutine != thread

Reply:
on second thought I think we need both @cjwagner, the first wg entry guarantees we run this handler, we also need to add more for each of the goroutines spawned as part of it. (so don't move the add into here, but do add more add / done in `handleGenericComment`

Reply:
Good catch, 72cddfc incorporates waiting for all goroutines spawned by handlers